### PR TITLE
[ERM UI] Add downstream domain UI work

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

[JIRA Ticket](https://dimagi-dev.atlassian.net/browse/SS-103)

**Review commit by commit**

This PR is responsible for adding the downstream domain button workflow to the Linked Projects page. This includes the following:

- Add Downstream Domain button and other minor UI tweaks to the Linked Projects page
- Passing in a list of available domains to link with when rendering the django view
- Trigger a modal to select from a list of available domains to link to. Currently only supports linking one at a time.
- Manually managing the list of both available domains and linked domains on the frontend to avoid excessive ajax calls

I left a todo in here to make an rmi call to add the new downstream domain which I will tackle next. This leaves the page in a bit of an incomplete state in that from the user's perspective it looks like they've added a domain, but they will be unable to remove it via the Remove Link button. However a simple refresh will remove them since no domains are actually being linked yet.

Screen Shot 2021-06-02 at 3 57 30 PM
Screen Shot 2021-06-02 at 3 57 37 PM
Screen Shot 2021-06-02 at 3 57 50 PM

## Feature Flag

`LINKED_DOMAINS`
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Added automated tests for the method to obtain a list of available domains to link with.
### QA Plan

Will run through QA before merging to master on parent branch.
### Safety story

Locally tested. This is not in a complete state and I have no intention of merging this PR into master as it is now.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/linked_domain/templates/linked_domain/domain_links.html [hunk 1]
```diff
@@ -79,6 +79,18 @@ <h2>{% trans "Project Link" %}</h2>  {# header for the only tab that will be dis
       {% endif %}
       <div class="tab-pane fade{% if not is_linked_domain %} in active{% endif %}" id="tabs-manage-downstream">
         <div id="ko-tabs-manage-downstream" class="ko-template">
+          <h3>{% trans "Manage Downstream Projects" %}</h3>
+          <p>{% trans "This project is an upstream project space for the projects listed below." %}</p>
+          <p><a href="#">{% trans "Learn more" %}</a>{% trans " about Linked Projects." %}</p>
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/linked_domain/templates/linked_domain/domain_links.html:84]: `target="_blank"` is also good for help links.

**Review by orangejenny — APPROVED** [review]
Just minor comments, looks good.

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js [hunk 2]
```diff
@@ -68,6 +69,7 @@ hqDefine("linked_domain/js/domain_links", [
         }
 
         self.domain_links = ko.observableArray(_.map(data.linked_domains, DomainLink));
+        self.newlyAddedDomainLinks = ko.observableArray([]);
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:72]: This is fair. The idea is to add new domains to the top of the table, but I could achieve that with using `unshift` on the `self.domain_links` array couldn't I?

**Review by orangejenny — APPROVED** [review]

**Review by biyeun — APPROVED** [review]
Looks great! 🎉

**Review by orangejenny — APPROVED** [review]

**Pull request merged**